#! /bin/sh /usr/share/dpatch/dpatch-run
## 80_killall_sched.dpatch by Petter Reinholdtsen
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Force the kernel to reschedule after killing processes.  Patch
## DP: from SuSe.

@DPATCH@
Index: sysvinit/src/killall5.c
===================================================================
--- sysvinit/src/killall5.c	(revisjon 64)
+++ sysvinit/src/killall5.c	(arbeidskopi)
@@ -618,5 +618,8 @@
 	/* Done. */
 	closelog();
 
+	/* Force the kernel to run the scheduler */
+	usleep(1);
+
 	return 0;
 }
